kernel/safe_typing.ml

```diff
@@ -452,22 +456,25 @@ let safe_push_named d env =
   Environ.push_named d env
 
 let push_named_def (id,de) senv =
-  let sections = Section.push_local senv.sections in
+  let sections = get_section senv.sections in
+  let sections = Section.push_local sections in
   let c, r, typ = Term_typing.translate_local_def senv.env id de in
   let x = Context.make_annot id r in
   let env'' = safe_push_named (LocalDef (x, c, typ)) senv.env in
-  { senv with sections; env = env'' }
+  { senv with sections=Some sections; env = env'' }
```

Comment:
Spaces are not a threatened species, please put a pair between equality, as well as everywhere below. Even better, reuse the punning and leave that line unchanged, writing above instead `let sections = Some (Section.push_local sections) in`.